src/Table2Beta/Table.tsx

```diff
@@ -221,11 +228,12 @@ export class Table2Beta extends React.Component<Props, State> {
         newSelectedRows.delete(item);
       }
     });
-    if (newSelectedRows.size !== selectedRows.size) {
+    const newAllSelected = (allRows && newSelectedRows.size === allRows.length) || false;
```

Comment:
Definitely a nit, but to cast to bool in JS it's most common to use `!!`:
```suggestion
    const newAllSelected = !!(allRows && newSelectedRows.size === allRows.length);
```